tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -99,7 +100,7 @@ func (s *stresser) run(ctx context.Context, kvc pb.KVClient) {
 	defer s.wg.Done()
 
 	for {
-		if err := s.rateLimiter.Wait(ctx); err == context.Canceled {
+		if err := s.rateLimiter.Wait(ctx); err == context.Canceled || ctx.Err() == context.Canceled {
```

Comment:
I found bunch of cases returning `nil` even after we cancel the context, which were causing inconsistent revision while comparing revisions.

Think it returns `nil` from timer channel in `rate` package.

https://github.com/golang/time/blob/master/rate/rate.go#L249-L251